File: core/src/main/java/io/github/msameer0/rhythmicrush/game/gameplay/interactables/portals/MiniPortal.java
package io.github.msameer0.rhythmicrush.game.gameplay.interactables.portals;

import io.github.msameer0.rhythmicrush.game.registries.Registry;

/**
 * A portal that toggles the player's mini state upon contact.
 */
@Registry(id = "mini_portal")
public class MiniPortal extends AbstractPortal {

    public MiniPortal(float x, float y) {
        super(x, y);
        this.type = PortalType.MINI;
    }

    public MiniPortal() {
        super();
        this.type = PortalType.MINI;
    }

    @Override
    public MiniPortal init(float x, float y) {
        super.init(x, y);
        return this;
    }
}

File: core/src/main/kotlin/io/github/msameer0/rhythmicrush/game/gameplay/interactables/portals/MiniPortal.kt
package io.github.msameer0.rhythmicrush.game.gameplay.interactables.portals

import io.github.msameer0.rhythmicrush.game.registries.Registry

@Registry(id = "mini_portal")
class MiniPortal : AbstractPortal {
    constructor(x: Float, y: Float) : super(x, y) {
        this.type = PortalType.MINI
    }

    constructor() : super() {
        this.type = PortalType.MINI
    }

    override fun init(x: Float, y: Float, rotation: Float): MiniPortal {
        super.init(x, y, rotation)
        this.type = PortalType.MINI
        return this
    }

    override fun init(x: Float, y: Float): MiniPortal {
        return init(x, y, 0f)
    }
}
